contracts/RewardPool.vy

```diff
@@ -35,46 +29,44 @@ event Claimed:
     claim_epoch: uint256
     max_epoch: uint256
 
+event AllowedToRelock:
+    user: indexed(address)
+    relocker: indexed(address)
+    allowed: bool
 
 struct Point:
     bias: int128
     slope: int128  # - dweight / dt
     ts: uint256
     blk: uint256  # block
 
+struct LockedBalance:
+    amount: uint256
+    end: uint256
 
-TWO_WEEKS: constant(uint256) = 14 * 86400
+
+WEEK: constant(uint256) = 7 * 86400
 TOKEN_CHECKPOINT_DEADLINE: constant(uint256) = 86400
 
+YFI: immutable(ERC20)
+VEYFI: immutable(VotingYFI)
+
 start_time: public(uint256)
 time_cursor: public(uint256)
 time_cursor_of: public(HashMap[address, uint256])
 user_epoch_of: public(HashMap[address, uint256])
+allowed_to_relock: public(HashMap[address, HashMap[address, bool]])  # user -> relocker -> allowed
 
 last_token_time: public(uint256)
-tokens_per_week: public(uint256[1000000000000000])
+tokens_per_week: public(HashMap[uint256, uint256])
 
-voting_escrow: public(address)
-token: public(address)
 total_received: public(uint256)
 token_last_balance: public(uint256)
-
-ve_supply: public(uint256[1000000000000000])  # VE total supply at week bounds
-
-admin: public(address)
-future_admin: public(address)
-emergency_return: public(address)
-is_killed: public(bool)
+ve_supply: public(HashMap[uint256, uint256])
 
 
 @external
-def __init__(
-    _voting_escrow: address,
-    _start_time: uint256,
-    _token: address,
-    _admin: address,
-    _emergency_return: address
-):
+def __init__(veyfi: address, start_time: uint256):
```

Comment:
ditto -- `veyfi` could just have type `VotingYFI` here to begin with, and skip the cast on line 83.

Reply:
great tip, thank you. fixed in https://github.com/yearn/veYFI/pull/169